contrib/utilities/detect_merges_in_feature_branch.sh

```diff
@@ -0,0 +1,27 @@
+#!/bin/sh
+## ------------------------------------------------------------------------
+##
+## SPDX-License-Identifier: LGPL-2.1-or-later
+## Copyright (C) 2025 by the deal.II authors
+##
+## This file is part of the deal.II library.
+##
+## Part of the source code is dual licensed under Apache-2.0 WITH
+## LLVM-exception OR LGPL-2.1-or-later. Detailed license information
+## governing the source code and code contributions can be found in
+## LICENSE.md and CONTRIBUTING.md at the top level directory of deal.II.
+##
+## ------------------------------------------------------------------------
+
+# 
+# This is a little script that checks if the git history of the 
+# feature branch is linear, 
+# i.e. no merges of branch 'master' into that branch are present.
+# 
+# NOTE: This script will do nothing in branch 'master' since the log will always be empty
+#       
+
+if [[ $(git log master.. --merges) ]]; then
+  echo "There are merge commits in this feature branch!"
```

Comment:
It is not uncommon to have merges on feature branches, and they are ok too. It's just that you shouldn't merge from `master`. Can you make that clear in the error message?